community/kernel/src/main/java/org/neo4j/kernel/monitoring/Monitors.java

```diff
@@ -289,6 +289,8 @@ private void invokeMonitorListeners( Object proxy, Method method, Object[] args
                     }
                     catch ( Throwable e )
                     {
+                        e.printStackTrace();
+                        throw new RuntimeException( e );
                         // TODO: something?! Ignore, rethrow etc.
```

Comment:
This affects a lot of call-sites. Letting the exceptions bubble through monitor invocation exposes a lot to new failure modes that they perhaps are not prepared to handle, so I think we should just log the exception instead. Via a logger, of course.

Reply:
Addressed with an optional LogProvider in constructor and logging with message.Thanks